zipline/assets/futures.py

```diff
@@ -0,0 +1,165 @@
+from pandas import Timestamp, Timedelta
+from pandas.tseries.tools import normalize_date
+
+
+class FutureChain(object):
+    """ Allows users to look up future contracts.
+
+    Parameters
+    ----------
+    asset_finder : AssetFinder
+        An AssetFinder for future contract lookups, in particular the
+        AssetFinder of the TradingAlgorithm instance.
+    get_datetime : function
+        A function that returns the simulation datetime, in particular
+        the get_datetime method of the TradingAlgorithm instance.
+    root_symbol : str
+        The root symbol of a future chain.
+    as_of_date : pandas.Timestamp, optional
+        Date at which the chain determination is rooted. I.e. the
+        existing contract that expires first after (or on) this date is
+        the primary contract, etc. If not provided, the current
+        simulation date is used as the as_of_date.
+
+    Attributes
+    ----------
+    root_symbol : str
+        The root symbol of the future chain.
+    as_of_date
```

Comment:
Since this is a property and has it's own docstring the numpy conventions suggest to just include it by name
https://github.com/numpy/numpy/blob/master/doc/HOWTO_DOCUMENT.rst.txt#class-docstring